cmd/kubeadm/app/images/images_windows.go

```diff
@@ -0,0 +1,29 @@
+// +build windows
+
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package images
+
+import (
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	"k8s.io/kubernetes/cmd/kubeadm/app/constants"
+)
+
+// GetPauseImage returns the image for the "pause" container
+func GetPauseImage(cfg *kubeadmapi.ClusterConfiguration) string {
+	return GetGenericImage("mcr.microsoft.com/oss/kubernetes", "pause", constants.PauseVersion)
```

Comment:
This is likely to cause some problems with air gaped environments. Suppose the following:
- Users are in an air gaped environment
- Users have a local container registry with all Windows images in it.
- They have overwritten the kubeadm config to use that container registry.
In that case we are going to pull everything from the local registry except the Windows `pause` image which is going to be hardwired to `mcr.microsoft.com/oss/kubernetes` and will require internet access or manual intervention on each Windows node.

The right thing to do here is to have pause image as a fat multi-platform image in the GCR repo and not have separate version of this function. Another possibility is to switch to `mcr.microsoft.com/oss/kubernetes` as the default image repository for Windows nodes. This would require the addition of kube-proxy to it (when kube-proxy starts to be executed as a DaemonSet).
A third possibility is to have an ugly hack in kubeadm, that is going to check if the image repository is the default gcr one and replace it with `mcr.microsoft.com` only in the case of `pause`.

@neolit123 WDYT?

Reply:
i forgot about air gapped use case and given the complications around multi-OS manifests and licensing, i think we might want to go with the hack solution to solve the problem for `kubeadm config images` too.

Reply:
Another possibility is to stop managing `pause` from kubeadm. It's really a CRI thing and we can add it as a requirement for folks to pre-pull manually after they install Docker/containerd/CRI-O.
That way we can get the same experience for all container runtimes.

Reply:
we'd need a GA deprecation announcement of sorts to allow smoother transition for that.
it will certainly break the existing user base that assumes kubeadm will pull pause for them on new nodes.